src/validators/object.ts

```diff
@@ -54,6 +54,70 @@ export class FieldsValidator<T> implements ValidatorInstance<Indexable<T>> {
   }
 }
 
+/**
+ * @api primitive
+ *
+ * The class that powers the `allFieldsPresent()` validator function.
+ */
+export class AllFieldsPresentValidator<T> implements ValidatorInstance<Indexable<T>> {
+  static validatorName = "all-fields-present";
+
+  constructor(
+    protected env: Environment,
+    protected descriptors: Dict<ValidationDescriptor<T>>
+  ) { }
+
+  run(value: Indexable<T>): Task<ValidationError[]> {
+    return new Task(async () => {
+      let errors: ValidationError[] = [];
+      let valueKeys = Object.keys(value);
+
+      for (let [key] of entries(this.descriptors)) {
+        let index = valueKeys.indexOf(key.toString());
+        if (index === -1) {
+          // descriptor field is not present in the value
+          errors.push({ path: [key.toString()], message: { name: "present", details: null } });
```

Comment:
ditto

Reply:
Also, the absence/presence validator does `{ name: "type", details: "absent/present" }`, we probably want to match that